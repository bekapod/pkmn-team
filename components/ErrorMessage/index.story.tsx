import { storiesOf } from "@storybook/react";
import React from "react";
import ErrorMessage from ".";

storiesOf("ErrorMessage", module)
  .add(
    "default",
    (): JSX.Element => <ErrorMessage>This is an error message.</ErrorMessage>
  )
  .add(
    "big",
    (): JSX.Element => (
      <ErrorMessage isBig>This is an error message.</ErrorMessage>
    )
  );
