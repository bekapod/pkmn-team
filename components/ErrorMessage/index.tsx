import React, { ReactNode, ReactNodeArray } from "react";
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

interface Props {
  isBig?: boolean;
  color?: string;
  children: ReactNode | ReactNodeArray;
}

const StyledError = styled.div`
  color: ${({ color }: Props) => color || variables.colors.error};
  font-weight: 400;

  ${({ isBig }: Props) =>
    isBig
      ? `
    margin: ${variables.spacing.xl}px 0;
    font-size: ${variables.fontSizes.md}px;
    text-align: center;
  `
      : ""}

  &::before {
    content: "";
    width: 100%;
  }
`;

const ErrorMessage = ({ children, ...props }: Props): JSX.Element => (
  <StyledError role="alert" {...props}>
    {children}
  </StyledError>
);

export default ErrorMessage;
