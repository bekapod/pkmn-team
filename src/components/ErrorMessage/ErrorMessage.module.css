.base {
  color: var(--color-error);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  font-weight: 400;

  &::before {
    content: '';
    width: 100%;
  }
}

.is-big {
  margin: var(--spacing-xl) 0;
  font-size: var(--font-size-lg);
  line-height: var(--line-height-lg);
  text-align: center;
}
