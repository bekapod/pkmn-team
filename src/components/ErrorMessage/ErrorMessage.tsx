import { FunctionComponent } from 'react';
import styled from 'styled-components/macro';

export type ErrorMessageProps = {
  isBig?: boolean;
  color?: string;
};

const StyledError = styled.div<ErrorMessageProps>`
  color: ${({ color }): string => color || 'var(--color-error)'};
  font-weight: 400;
  ${({ isBig }): string =>
    isBig
      ? `
    margin: var(--spacing-xl) 0;
    font-size: var(--font-size-md);
    line-height: var(--spacing-md);
    text-align: center;
  `
      : ''}

  &::before {
    content: '';
    width: 100%;
  }
`;

export const ErrorMessage: FunctionComponent<ErrorMessageProps> = ({
  children,
  ...props
}) => (
  <StyledError role="alert" {...props}>
    {children}
  </StyledError>
);
