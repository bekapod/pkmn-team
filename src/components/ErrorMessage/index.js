// @flow
import * as React from "react";
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

type Props = {
  children: React.ChildrenArray<React.Element<*>> | string
};

const StyledError = styled.div`
  color: ${variables.colors.error};
  font-weight: 700;

  &::before {
    content: "";
    width: 100%;
  }
`;

const ErrorMessage = ({ children }: Props) => (
  <StyledError role="alert">{children}</StyledError>
);

export default ErrorMessage;
