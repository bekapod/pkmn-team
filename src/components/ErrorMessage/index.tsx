import React, { ReactNode, ReactNodeArray } from "react";
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

interface IProps {
  isBig?: boolean;
  children: ReactNode | ReactNodeArray;
}

const StyledError = styled.div`
  color: ${variables.colors.error};
  font-weight: 700;

  ${({ isBig }: IProps) =>
    isBig
      ? `
    margin: ${variables.spacing.xl}px 0;
    font-size: ${variables.fontSizes.md}px;
    text-align: center;
  `
      : ""}

  &::before {
    content: "";
    width: 100%;
  }
`;

const ErrorMessage = ({ isBig, children }: IProps) => (
  <StyledError isBig={isBig} role="alert">
    {children}
  </StyledError>
);

export default ErrorMessage;
