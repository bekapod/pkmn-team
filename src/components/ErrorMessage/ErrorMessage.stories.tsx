import { Meta } from '@storybook/react/types-6-0';
import { ErrorMessage, ErrorMessageProps } from './ErrorMessage';

export default {
  title: 'Components/ErrorMessage',
  component: ErrorMessage
} as Meta<ErrorMessageProps>;

export const Standard = (args: ErrorMessageProps): JSX.Element => (
  <ErrorMessage {...args}>This is an error message.</ErrorMessage>
);
