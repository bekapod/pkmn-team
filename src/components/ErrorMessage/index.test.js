// @flow
import React from "react";
import renderer from "react-test-renderer";
import ErrorMessage from ".";

describe("<ErrorMessage />", () => {
  describe("when text is passed", () => {
    it("renders without crashing", () => {
      const tree = renderer.create(
        <ErrorMessage>This is an error</ErrorMessage>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe("when a html element is passed", () => {
    const tree = renderer.create(
      <ErrorMessage>
        <span>This is an error in a span</span>
      </ErrorMessage>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe("when a react component is passed", () => {
    const Component = () => <div>This is an error in a component</div>;
    const tree = renderer.create(
      <ErrorMessage>
        <Component />
      </ErrorMessage>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
