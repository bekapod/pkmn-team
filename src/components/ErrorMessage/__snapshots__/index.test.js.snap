// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="ErrorMessage__StyledError-sc-1q305d-0 cnWPsI"
  role="alert"
>
  <span>
    This is an error in a span
  </span>
</div>
`;

exports[` 2`] = `
<div
  className="ErrorMessage__StyledError-sc-1q305d-0 cnWPsI"
  role="alert"
>
  <div>
    This is an error in a component
  </div>
</div>
`;

exports[`<ErrorMessage /> when text is passed renders without crashing 1`] = `
<div
  className="ErrorMessage__StyledError-sc-1q305d-0 cnWPsI"
  role="alert"
>
  This is an error
</div>
`;
